src/sagemaker/predictor.py

```diff
@@ -109,23 +110,42 @@ def _delete_endpoint_config(self):
         """Delete the Amazon SageMaker endpoint configuration
 
         """
-        endpoint_description = self.sagemaker_session.sagemaker_client.describe_endpoint(EndpointName=self.endpoint)
-        endpoint_config_name = endpoint_description['EndpointConfigName']
-        self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
+        self.sagemaker_session.delete_endpoint_config(self._endpoint_config_name)
```

Comment:
I think it would be less confusing to the have the _endpoint_config_name class attribute to be introduced in the __init__.